controllers/operatorpolicy_controller.go

```diff
@@ -278,57 +404,42 @@ func (r *OperatorPolicyReconciler) updatePolicyStatus(
 // with the apiversion and kind in preparation for resource creation
 func buildSubscription(
 	policy *policyv1beta1.OperatorPolicy,
-) *operatorv1alpha1.Subscription {
+) (*operatorv1alpha1.Subscription, error) {
 	subscription := new(operatorv1alpha1.Subscription)
 
-	subscription.SetGroupVersionKind(subscriptionGVK)
-	subscription.ObjectMeta.Name = policy.Spec.Subscription.Package
-	subscription.ObjectMeta.Namespace = policy.Spec.Subscription.Namespace
-	subscription.Spec = policy.Spec.Subscription.SubscriptionSpec.DeepCopy()
+	sub := make(map[string]interface{})
 
-	OpLog.Info("Creating the subscription", "kind", subscription.Kind,
-		"namespace", subscription.Namespace)
+	err := json.Unmarshal(policy.Spec.Subscription.Raw, &sub)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshalling subscription: %w", err)
+	}
 
-	return subscription
-}
+	ns, ok := sub["namespace"].(string)
+	if !ok {
+		return nil, fmt.Errorf("namespace is required in spec.subscription")
+	}
 
-// Sets the compliance of the policy
-func (r *OperatorPolicyReconciler) setCompliance(
-	ctx context.Context,
-	policy *policyv1beta1.OperatorPolicy,
-	compliance policyv1.ComplianceState,
-) {
-	policy.Status.ComplianceState = compliance
+	spec := new(operatorv1alpha1.SubscriptionSpec)
 
-	err := r.updatePolicyStatus(ctx, policy)
+	err = json.Unmarshal(policy.Spec.Subscription.Raw, spec)
```

Comment:
I don't think this will catch invalid/unknown fields. It looks like the unstructured package's converter to typed has some unknown field validation.